pkg/datapath/garp/events.go

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package garp
+
+import (
+	"context"
+	"net/netip"
+
+	"github.com/cilium/workerpool"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+var nodePodsCell = cell.Provide(func(lc hive.Lifecycle, c k8sClient.Clientset) (resource.Resource[*corev1.Pod], error) {
+	if !c.IsEnabled() {
+		return nil, nil
+	}
+
+	lw := utils.ListerWatcherWithFields(utils.ListerWatcherFromTyped[*corev1.PodList](
+		c.CoreV1().Pods("")), fields.OneTermEqualSelector("spec.nodeName", nodeTypes.GetName()),
+	)
+	return resource.New[*corev1.Pod](lc, lw), nil
+})
+
+// EventProcessorCell processes k8s pod events for the current node and
+// determines if a Gratuitous ARP packet needs to be sent.
+var EventProcessorCell = cell.Module(
+	"garp-event-processor",
+	"GARP event processor",
```

Comment:
Use a descriptive title that someone unfamiliar with this can understand, e.g. "Sends gratuitous ARP packet for each local pod". The title is always associated with the module id, e.g. it'll look like "garp-event-processor: Sends gratuitous ARP packets for each local pod" (once we implement the module dependency diagram generation anyway)